Library/Homebrew/command_registry.rb

```diff
@@ -0,0 +1,19 @@
+# typed: strong
+# frozen_string_literal: true
+
+module Homebrew
+  module CommandRegistry
```

Comment:
The existence of this module might be overkill, I could go either way. I originally had the registry implemented in `AbstractCommand`, but thought the registry polluted the concept of the superclass: https://github.com/Homebrew/brew/pull/16815/commits/4295a333d0051024677bf953413b492349573a20